runtime/drivers/admin/admin.go

```diff
@@ -279,9 +283,45 @@ func (h *Handle) cloneOrPull(ctx context.Context) error {
 			err = yaml.Unmarshal(rawYaml, yml)
 			if err == nil {
 				h.ignorePaths = yml.IgnorePaths
+				h.cachedPaths = yml.PublicPaths
 			}
 		}
 
+		cache := make(map[string][]byte)
+		for _, p := range h.cachedPaths {
+			p = filepath.Join(h.projPath, p)
+			_, err = os.Stat(p)
+			if err != nil {
+				if os.IsNotExist(err) {
+					continue
+				}
+				return nil, err
+			}
+
+			err := filepath.Walk(p, func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+				if !info.IsDir() {
+					b, err := os.ReadFile(path)
+					if err != nil {
+						return err
+					}
+
+					rel, err := filepath.Rel(h.projPath, path)
+					if err != nil {
+						return err
+					}
+					cache[strings.TrimLeft(rel, "/")] = b
+				}
+				return nil
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+		h.assetsCache = cache
```

Comment:
This is not a useful cache. The place where caching is nice is to be able to serve assets immediately when the runtime starts, even if Github takes a few minutes to clone. This means the cached files would need to be stored on the persistent disk, and be readable before the first call to `cloneOrPull`.

Also, we should not eagerly cache the assets in memory – it can lead to excessive memory usage if there are larger files. Relying on the OS file system page cache is more than adequate for our performance requirements here.

Reply:
I wonder how a user can get into a state where there's cache on disk but no cloned repo? Because cache can be created from a clone repo.

And why would user like to delete a cloned repo leaving just the cache on disk?

Reply:
The repo is currently cloned into a temp directory which will not exist after a process restart: https://github.com/rilldata/rill/blob/1983fb1194d9f358eb5321534b15a4a824b0910c/runtime/drivers/admin/admin.go#L356.

It might be fine to cache the full repo on the persistent disk – that just isn't what it's doing now.